app.py

```diff
@@ -45,10 +45,13 @@ def inner_decorator(func):
         @wraps(func)
         def wrapped_func(*args, **kwargs):
             token = None
-            if request.method == 'GET':
+            token = request.cookies.get('token')
+            if token is None:
                 token = request.args.get('token')
-            elif request.method == 'POST':
-                token = request.values.get('token')
+                if token is not None:
```

Comment:
You are first checking if there is a cookie and in that case the cookie value is used. But if the cookie-token has expired then `verify_token` will fail and the user will be redirected to `login.datasektionen.se/login?...`. After that the user will be redirected back to zfinger with a `?token=` parameter, but since we already have a token in the cookie the new token will be ignored.

Instead you should first check if there is a token parameter and in that case set it in the cookie and redirect back to the current page.

Reply:
This one I'm unsure if I interpreted correctly